vhdl_parser/src/tokenizer.rs

```diff
@@ -401,6 +401,14 @@ pub struct Token {
     pub kind: Kind,
     pub value: Value,
     pub pos: SrcPos,
+    pub leading_comments: Vec<Comment>,
+    pub trailing_comment: Option<Comment>,
```

Comment:
Maybe the comments should be put in a struct and put in an Option<Box<Commens>> to minimize Token sizes when there are no comments.

Reply:
Also consider move_after(&token) when there are trailing comments.

Reply:
I put the token comments in an Option<Box> as suggested.

I've modified move_after so I can use it with trailing_comments,  but I haven't pushed that yet.  I get an error about mutable and immutable borrows, which I need to work out.  I'm enjoying rust, but haven't quite got the hang of it yet!